packages/peregrine/lib/talons/WishlistPage/useActionMenu.js

```diff
@@ -0,0 +1,93 @@
+import { useCallback, useState } from 'react';
+import { useMutation } from '@apollo/client';
+import DEFAULT_OPERATIONS from './wishlist.gql';
+import mergeOperations from '../../util/shallowMerge';
+
+const dialogs = {
+    NONE: 1,
+    LIST_ACTIONS: 2,
+    EDIT_WISHLIST: 3
+};
+
+/**
+ * @function
+ *
+ * @param {{id}} props
+ * @param {ID} props.id - The unique identifier of the wish list
+ * @param {Object} props.operations - GraphQL operations to be run by the talon.
+ *
+ * @returns {ActionMenuProps}
+ */
+export const useActionMenu = (props = {}) => {
+    const { id } = props;
+    const operations = mergeOperations(DEFAULT_OPERATIONS, props.operations);
+    const { getCustomerWishlistQuery, updateWishlistMutation } = operations;
+    const [currentDialog, setCurrentDialog] = useState(dialogs.NONE);
+
+    const handleActionMenuClick = useCallback(() => {
+        setCurrentDialog(dialogs.LIST_ACTIONS);
+    }, []);
+
+    const handleHideDialogs = useCallback(() => {
+        setCurrentDialog(dialogs.NONE);
+    }, []);
+
+    const listActionsIsOpen = currentDialog === dialogs.LIST_ACTIONS;
+    const editFavoritesListIsOpen = currentDialog === dialogs.EDIT_WISHLIST;
+
+    const handleShowEditFavorites = useCallback(() => {
+        setCurrentDialog(dialogs.EDIT_WISHLIST);
+    }, []);
+
+    const [
+        updateWishlist,
+        { error: updateWishlistErrors, loading: isEditInProgress }
+    ] = useMutation(updateWishlistMutation);
+
+    const handleEditWishlist = useCallback(
+        async data => {
+            try {
+                await updateWishlist({
+                    variables: {
+                        name: data.name,
+                        visibility: data.visibility,
+                        wishlistId: id
+                    },
+                    refetchQueries: [{ query: getCustomerWishlistQuery }],
+                    awaitRefetchQueries: true
+                });
+                await setCurrentDialog(dialogs.NONE);
```

Comment:
As far as I know this is async but awaiting doesn't do anything. May as well remove the `await`.